server/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/globalstate/SnapshotGlobalStateRequest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.snapshots.globalstate;
+
+import org.elasticsearch.action.ActionRequestValidationException;
+import org.elasticsearch.action.support.master.MasterNodeRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.snapshots.SnapshotId;
+import org.elasticsearch.tasks.CancellableTask;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.tasks.TaskId;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static org.elasticsearch.action.ValidateActions.addValidationError;
+
+/**
+ * Get snapshot global state status request
+ */
+public class SnapshotGlobalStateRequest extends MasterNodeRequest<SnapshotGlobalStateRequest> {
+
+    private String repository = "_all";
+
+    private String snapshot;
```

Comment:
Let's make these fields final, they can be set in the constructor:

```suggestion
    private final String repository;
    private final String snapshot;
```

Reply:
Made these fields final as suggested.